examples/TwoD_LidCavity.jl

```diff
@@ -0,0 +1,57 @@
+using WaterLily
+include("TwoD_plots.jl")
+
+mag(I,u) = √sum(ntuple(i->0.25*(u[I,i]+u[I+δ(i,I),i])^2,length(I)))
+
+# import explicitly BC function and overwrite
+function BC_lid!(a,A)
+    N,n = WaterLily.size_u(a)
+    for j ∈ 1:n, i ∈ 1:n
+        if i==1 && j==2 # lid, Dirichlet cannot be imposed, must interpolate u[i,j+1,1]+u[i,j]/2 = uBC
+            @WaterLily.loop a[I,i] = 2.00-a[I-δ(j,I),i] over I ∈ WaterLily.slice(N,N[j],j)
+            @WaterLily.loop a[I,i] = -a[I+δ(j,I),i] over I ∈ WaterLily.slice(N,1,j)
+        elseif i==j # Normal direction, homoheneous Dirichlet
+            @WaterLily.loop a[I,i] = 0.0 over I ∈ WaterLily.slice(N,1,j)
+            @WaterLily.loop a[I,i] = 0.0 over I ∈ WaterLily.slice(N,N[j],j)
+        else  # Tangential directions, interpolate ghost cell to homogeneous Dirichlet
+            @WaterLily.loop a[I,i] = -a[I+δ(j,I),i] over I ∈ WaterLily.slice(N,1,j)
+            @WaterLily.loop a[I,i] = -a[I-δ(j,I),i] over I ∈ WaterLily.slice(N,N[j],j)
+        end
+    end
+end
+@fastmath function WaterLily.mom_step!(a::Flow{N},b::AbstractPoisson) where N
```

Comment:
It's nice that this works. But it would be better to have an interface so a user can pass a BC function and we would just substitute it in `mom_step!` internally.